/*
    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        https://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

syntax = "proto3";

import "JavaUdfDescriptorMessage.proto";

package NES;

option java_package = "stream.nebula.protobuf";
option java_multiple_files = true;

message RegisterJavaUdfRequest {
  string udf_name = 1;
  JavaUdfDescriptorMessage java_udf_descriptor = 2;
}

message GetJavaUdfDescriptorResponse {
  bool found = 1;
  optional JavaUdfDescriptorMessage java_udf_descriptor = 2;
}
